core/src/test/java/nu/yona/server/messaging/service/MessageServiceTest.java

```diff
@@ -207,6 +207,7 @@ public void sendDirectMessage_userCreatedOnBuddyRequest_doesNotSendSms()
 	public void prepareMessageCollection_transferringDirectMessagesToAnonymousDestination_sendsFirebaseMessage()
 	{
 		Message message = Mockito.mock(Message.class);
+		when(message.duplicate()).thenReturn(message);
```

Comment:
Can we not add some more asserts about what `prepareMessageCollection` does now?

Reply:
This change doesn't really change that, only the JPA-related behavior is changed, but that is hard to unit test. Given the urgency of this fix, I'm not going to spend time on this now.